File: workspace/android_studio/Test/app/src/main/java/com/example/yangqiang/test/activity/ThirdActivity.java
package com.example.yangqiang.test.activity;

import android.content.Intent;
import android.support.v4.app.NavUtils;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

import com.example.yangqiang.test.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class ThirdActivity extends AppCompatActivity implements View.OnClickListener {

    @BindView(R.id.btn_third) Button mBtnAnim;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_third);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.btn_third)
    public void onClick() {
        startActivity(new Intent(ThirdActivity.this, MainActivity.class));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        finish();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_third:
                break;
        }
    }
}

File: android/ARouterDemo/app/src/main/java/com/my/arouterdemo/activity/ThirdActivity.kt
package com.my.arouterdemo.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.alibaba.android.arouter.facade.annotation.Route
import com.my.arouterdemo.R
import com.my.common.Const

@Route(path = Const.PATH_THIRD_ACT)
class ThirdActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_third)
    }
}